addon-test-support/-private/utils.js

```diff
@@ -0,0 +1,14 @@
+import Ember from 'ember';
+
+export function checkMatcher(message, matcher) {
+  if (typeof matcher === 'string') {
+    return message.includes(matcher);
```

Comment:
I'm more 😬  about this because it would almost surely break existing users. And in some cases the the deprecation that's expected could contain dynamic data (e.g. ember component IDs, model IDs).

I think the decision comes down to what is most likely to be desired for users. Because of RegExp, you can always do `/^XYZ$/` for exact match and `/.*XYZ.*/` for wildcard match.

Let's talk it out. What's your thinking behind exact matching?